### ISSUE TITLE
gradle aggregate : mixed reports ? 

### ISSUE THREAD

**Issue by mmanach** [serenity-bdd/serenity-core#976]
Hello, 

Just to let you know of a strange behavior with gradle on Windows 10.

1. I run tests of project A from IntelliJ.
2. Then I check the serenity report, all ok
3. (I noticed that a java process is still here after this execution, seems to be the gradle daemon)
4. I run tests of project B from IntelliJ
5. Then I check the serenity report, and some part are mixed with the report of project A
6. To have a clean report, I need to kill this gradle daemon

![mixed reports](https://user-images.githubusercontent.com/29480659/31712724-2f4b5eba-b3fc-11e7-852c-275d3072422e.png)

**Comment by wakaleo** [comment]
This could be due to Windows file locks. Can you produce two projects that reproduce the issue?

**Comment by yemo** [comment]
I have the same problem when executing gradle in command line to run test.  Not sure how 'aggregate' works and where it collects previous run project report files. two projects are in different folders, the report data source may come from memory.

**Comment by wakaleo** [comment]
The aggregate task collects the JSON test outcomes from the output directory (usually target/site/serenity) and converts them into an aggregate report. The only way I could imagine it generating test results from two projects is if the two projects produce results to the same target directory.

**Comment by yemo** [comment]
First time I seen this problem is on our Bamboo server. I don't think Bamboo agent check out two projects into same directory. I checked the Bamboo agent, two projects in different directories, and their outcomes in their own directories target folder.
It also happen on my local machine occasionally. After I ran one project, I ran the other project. Even I deleted the second project's target folder, and ran again, the first project info still showed on the second projects report (feature tab). It can be fixed by restart my local machine (kill gradle deamon), so I assumed aggregate task may also read some thing in the memory.

**Comment by andres-florit** [comment]
I experienced a different issue, but also produced by gradle daemon.

My issue was getting some previous erased capabilities or features in the generated report for the same project. I.e, if I had a feature A, then generated the report, then renamed the feature A to B and generated the report again, then I got both, A and B in some report tabs. Even if executed the _clean_ task or removed _target_, _build_, and _<project>/.gradle_ folders manually. Something was cached somewhere and I found it in my _~/.gradle/daemon/3.5_ folder. Removing that folder content before each test (besides executing _clean_ task) or **disabling gradle daemon** worked for me.

To disable gradle daemon, add the following line to the _~/.gradle/gradle.properties_

`org.gradle.daemon=false`

Environment: Ubuntu 16.04, Gradle 3.5, Java 1.8.0_101

To reproduce the issue, clone [journey planner](https://github.com/serenity-bdd/journey-planner) repo and play changing features and capabilities changing names and moving them in the hierarchy. Generate reports between changes. Use an installed gradle (not the wrapper) and local browser (no tested with selenium hub/grid). I think I experienced the same issue with the [serenity cucumber starter](https://github.com/serenity-bdd/serenity-cucumber-starter) repo too, didn't test it again if disabling the daemon fix it.
